File: Lua/selection_sort.lua
function selection_sort(arr)
    for i = 1, #arr - 1 do
        local min_idx = i
        for j = i + 1, #arr do
            if arr[j] < arr[min_idx] then
                min_idx = j
            end
        end
        arr[i], arr[min_idx] = arr[min_idx], arr[i]
    end
end

local arr = {5, 1, 4, 2, 8}
selection_sort(arr)
print(table.concat(arr, " "))

File: Python/selection_sort.py
def selection_sort(arr):
    n = len(arr)
    for i in range(n):
        min_idx = i
        for j in range(i + 1, n):
            if arr[j] < arr[min_idx]:
                min_idx = j
        arr[i], arr[min_idx] = arr[min_idx], arr[i]
    return arr

arr=[3, 7, 4, 8, 6, 2, 1, 5, 0, -5, -2]
print(selection_sort(arr))